### PR TITLE
feat(docs): reorganize top-level menus in online docs

### PR DESCRIPTION
**Pull request by danceratopz**
## 🗒️ Description
This PR:
- Reorganizes the top-level menus more logically and brings them more inline with the changes in this repo over the last year.
- Fixes the now incorrect language of "executing tests" in the context of "filling tests" :)

Preview available here: https://ethereum.github.io/execution-spec-tests/pr-929-preview/

## 🔗 Related Issues
None.

## ✅ Checklist
- [x] All: Set appropriate labels for the changes.
- [x] All: Considered squashing commits to improve commit history.
- [x] All: Added an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://ethereum.github.io/execution-spec-tests/main/tests/) are correctly formatted.

### CODE REVIEW

File: docs/filling_tests/index.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Filling Tests
+
+Execution of test cases against clients is a two-step process. First JSON test fixtures are generated from the Python test cases in this repository using `fill`. Then the resulting JSON fixtures can be executed against a client. The process of generating fixtures is often referred to as "filling" the tests.
+
```
Comment (marioevz) [inline docs/filling_tests/index.md:4]: We should add a warning here that filling tests is in some cases possible even on a faulty client (cases where we cannot do a post-check in the test, such as in some of the gas tests), which means the filling could succeed and an invalid test fixture could be created, so a client filling tests should not be considered as free of bugs.

**Review by marioevz — APPROVED** [review]
LGTM, just a minor suggestion we could apply before merging.

**Comment by marioevz** [comment]
Btw, rebasing to main should fix the CI issues hopefully.

**Review by spencer-tb — APPROVED** [review]

File: docs/filling_tests/index.md (hunk 1 above)
Comment (danceratopz) [reply docs/filling_tests/index.md:9]: Just added the top-level filler page overview including DIRE WARNINGS about relying only on `fill` for correctness.

Comment (danceratopz) [reply docs/filling_tests/index.md:9]: It's here:
https://ethereum.github.io/execution-spec-tests/main/filling_tests/

**Pull request merged**